spartan_parallel/src/random.rs

```diff
@@ -1,29 +1,33 @@
-use super::scalar::Scalar;
+use super::scalar::SpartanExtensionField;
 use super::transcript::ProofTranscript;
 use merlin::Transcript;
 use rand::rngs::OsRng;
 
 #[derive(Clone)]
-pub struct RandomTape {
+pub struct RandomTape<S: SpartanExtensionField> {
   tape: Transcript,
+  _phantom: S,
 }
 
-impl RandomTape {
+impl<S: SpartanExtensionField> RandomTape<S> {
```

Comment:
If you name this variable `Scalar` instead of `S` the diff becomes a lot smaller and easier to review.  (Similar in many other places as well.)